File: code/ch13/13.1.1.best_time_to_buy_and_sell_stock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_diff = 0
        for i in range(1, len(prices)):
            for j in range(0, i):
                if prices[i] - prices[j] > max_diff:
                    max_diff = prices[i] - prices[j]
        return max_diff

File: code/ch13/13.1.1.best_time_to_buy_and_sell_stock.cpp
class Solution {
    public:
        int maxProfit(vector<int>& prices) {
            int maxprofit = 0;

            for (int i = 0; i < prices.size(); i++) {
                for (int j = i + 1; j < prices.size(); j++) {
                    if (prices[j] - prices[i] > maxprofit)
                        maxprofit = prices[j] - prices[i];
                }
            }

            return maxprofit;
        }
};
